doc/group_instances.md

```diff
@@ -0,0 +1,269 @@
+### Group Instance & Group Instance IDs
+
+---
+
+It is recommended that you familiarise yourself with [*repeating routing rules*](repeating_routing_rules.md) before proceeding with this.
+
+Group Instance / Instance IDs are used to identify the instance of an answer acquired from a repeating group.
+
+#### Group Instance (Index)
+An answer can have multiple instances, the *group_instance* tracks this instance and is part of the [*Location*](../app/questionnaire/location.py) object which consists of *group_id*, *group_instance*, and the *block_id*.
+
+Currently, `group_instance` is always used as it is part of the Location. Any non-repeating answers will have a `group_instance` of `0` by default.
+
+##### Group Instance - Repeats
+
+On every request to `/questionnaire/<eq_id>/<form_type>/<collection_id>/`, we generate the full routing [*Path*](../app/questionnaire/path_finder.py) which holds a list of *Location* objects.
+In short, every block in every group is added to the path. Any block or group with skip conditions or routing rules which evaluates *True* is not added to the path.
+
+When generating the Path, for each group in the schema, we increment the *group_instance* based on the number of times the group repeats. The *no. of repeats* is determined based on the evaluation of the `repeat` rule on each group. 
+
+
+##### Example
+
+```json
+"routing_rules": [{
+    "repeat": {
+        "type": "until",
+        "when": [{
+            "id": "repeating-anyone-else",
+            "condition": "equals",
+            "value": "No"
+        }]
+    }
+}]
+```
+The *no. of repeats* we will have for the above routing rule is based on the number of answers we have with the answer id *repeating-anyone-else* to which the answer is not `"No"`. The *no. of repeats* is incremented for each answer we have with matching id until the routing rule evaluates to *True*.
+If there are no answers yet, it will repeat once.
+
+---
+
+#### Group Instance ID
+
+Unlike `group_instance`, the `group_instance_id` is used only when a *Location* has group dependencies. A `group_instance_id` is a *UUID4* string which represents the instance of an answer. Locations without dependencies are given the default value of `None` for the `group_instance_id`. 
```

Comment:
A high level definition of a group dependency would be useful here.

Reply:
The next paragraph does this?.

Reply:
As discussed IRL, most of this section relies on the idea of a dependency, but I don't think it's ever explained what makes one group dependent on another group / block